rust/arrow/src/compute/kernels/arithmetic.rs

```diff
@@ -536,7 +555,29 @@ where
     #[cfg(simd)]
     return simd_signed_unary_math_op(array, |x| -x, |x| -x);
     #[cfg(not(simd))]
-    return signed_unary_math_op(array, |x| -x);
+    return Ok(unary(array, |x| -x));
+}
+
+/// Raise array to the power of a scalar.
+pub fn pow_scalar<T>(
```

Comment:
Given that this function is [powf](https://doc.rust-lang.org/stable/std/primitive.f64.html#method.powf), WDYT about renaming it as such? We're going to also have `pow` for the integer types, `powi` to raise floats to an integer power. So we could preempt the possible rename.

I'll remove the `powf` in #9313 because this one has a SIMD implementation.

Reply:
I renamed to `powf_scalar`. I still have a `scalar` suffix distinct with a (future) kernel that is a binary function (i.e. array to the power of array).